dartagnan/src/main/java/com/dat3m/dartagnan/expression/integers/IntUnaryExpr.java

```diff
@@ -0,0 +1,61 @@
+package com.dat3m.dartagnan.expression.integers;
+
+import com.dat3m.dartagnan.expression.Expression;
+import com.dat3m.dartagnan.expression.ExpressionVisitor;
+import com.dat3m.dartagnan.expression.base.UnaryExpressionBase;
+import com.dat3m.dartagnan.expression.type.IntegerType;
+
+import java.math.BigInteger;
+
+import static com.google.common.base.Verify.verify;
+
+public class IntUnaryExpr extends UnaryExpressionBase<IntegerType, IntUnaryOp> {
+
+    public IntUnaryExpr(IntUnaryOp operator, Expression operand, IntegerType t) {
+        super(t, operator, operand);
```

Comment:
Shouldn't we check the operand is an integer?

Reply:
Yes, the signature is also wrong right now (the last parameter is obsolete now).